python/lsst/afw/cameraGeom/camera.py

```diff
@@ -28,24 +28,35 @@
 class Camera(DetectorCollection):
     """!A collection of Detectors that also supports coordinate transformation
     """
-    def __init__(self, name, detectorList, transformMap):
+    def __init__(self, name, detectorList, transformMap, pupilFactoryClass):
         """!Construct a Camera
 
         @param[in] name  name of camera
         @param[in] detectorList  a sequence of detectors in index order
         @param[in] transformMap  a CameraTransformMap whose native system is FOCAL_PLANE
             and that at least supports PUPIL coordinates
+        @param[in] pupilFactoryClass  a PupilFactory class for this camera
         """
         self._name = name
         self._transformMap = transformMap
         self._nativeCameraSys = self._transformMap.getNativeCoordSys()
+        self._pupilFactoryClass = pupilFactoryClass
         super(Camera, self).__init__(detectorList)
 
     def getName(self):
         """!Return the camera name
         """
         return self._name
 
+    def getPupilFactory(self, *args, **kwargs):
```

Comment:
tbh I don't recall why I used `*args`, `**kwargs`.  I don't actually envision any additional arguments though, so I just changed this to `visitInfo, pupilSize, npix` full stop.